src/store/reducers/favorites.js

```diff
@@ -1,33 +1,31 @@
-const initialState = {
-  songs: [],
+import { Map, List } from 'immutable';
+
+import {
+  INITIALIZE_FAVORITES, ADD_FAVORITE, REMOVE_FAVORITE, TOGGLE_IS_FAVORITE_PLAYING,
+} from '../constants';
+
+const initialState = Map({
+  songs: List(),
   isPlaying: false,
-};
+});
 
 const favoritesReducer = (state = initialState, { type, payload }) => {
   switch (type) {
-  case 'SET_FAVORITES':
-    return {
-      ...state,
-      songs: payload,
-    };
-
-  case 'ADD_FAVORITE':
-    return {
-      ...state,
-      songs: state.songs.concat(payload),
-    };
-
-  case 'REMOVE_FAVORITE':
-    return {
-      ...state,
-      songs: state.songs.filter((song) => song.song_name !== payload.song_name),
-    };
-
-  case 'TOGGLE_IS_PLAYING':
-    return {
-      ...state,
-      isPlaying: payload,
-    };
+  case INITIALIZE_FAVORITES:
+    return state.set('songs', payload);
+
+  case ADD_FAVORITE: {
+    const favoritesUpdated = state.get('songs').push(payload);
+    return state.set('songs', favoritesUpdated);
+  }
+
+  case REMOVE_FAVORITE: {
+    const favoritesUpdated = state.songs.filter((song) => song.song_name !== payload.song_name);
+    return state.set('songs', favoritesUpdated);
+  }
+
+  case TOGGLE_IS_FAVORITE_PLAYING:
+    return state.set('isPlaying', payload);
```

Comment:
I think makes more sense if the reducer toggles the current value. As the implementation is written right now it should be named `SET_` instead of `TOGGLE_`.